### ISSUE TITLE
Incorrect "JSC_USED_GLOBAL_THIS" warning on `this` in class field initializer

### ISSUE THREAD

**Issue by lauraharker** [google/closure-compiler#4230]
As noted in https://github.com/google/closure-compiler/issues/2731 - the compiler incorrectly reports "[JSC_USED_GLOBAL_THIS] dangerous use of the global this object" on the `this.y` reference below.

``` js
class Foo {
  x = () => this.y;
  y = 3;
}
```

`this` in a non-static class field initializer will refer to the class instance.

To reproduce -

[closure-compiler-debugger](https://closure-compiler-debugger.appspot.com/#input0%3Dclass%2520Foo%2520%257B%250A%2520%2520x%2520%253D%2520()%2520%253D%253E%2520this.y%253B%250A%2520%2520y%2520%253D%25203%253B%250A%257D%26input1%26conformanceConfig%26externs%26refasterjs-template%26CHECK_TYPES%3Dtrue%26REWRITE_MODULES_BEFORE_TYPECHECKING%3Dtrue%26CHECK_GLOBAL_THIS%3Dtrue%26CLOSURE_PASS%3Dtrue%26PARSE_TYPE_ANNOTATIONS%3Dtrue%26PRETTY_PRINT%3Dtrue)

**Comment by lauraharker** [comment]
This also repros when `this` is not wrapped in an arrow function.

**Comment by trevorade** [comment]
This issue was resolved with https://github.com/google/closure-compiler/commit/fc6b14e9f1192fc62d8ced28e740d54c6440b150
